### PR TITLE
Segmentation fix

### LINKED ISSUE #2269: run_asr_segmentation.sh stage 2 makes train directory invalid
background: trying to do SAD using [run_asr_segmentation.sh](https://github.com/kaldi-asr/kaldi/blob/master/egs/swbd/s5c/local/run_asr_segmentation.sh). 

Main training data is in $data_dir which is fixed/validated with utterance ids prefixed with speaker ids. 

[prepare_targets_gmm.sh](https://github.com/kaldi-asr/kaldi/blob/master/egs/wsj/s5/steps/segmentation/prepare_targets_gmm.sh) in stage 3 reguires both <data> and <whole-recording-data> to have the recording-id as speaker. 

For this reasosn, stage 2 of `run_asr_segmentation.sh ` tries to replace speaker ids in utt2spk with recording ids but it's an incomplete effort: utterance ids still include old speaker ids as prefix and this in return fails `validate_data_dir.sh` since new speaker ids (recording ids) are not prefix of utterance ids. 

Kaldi 5.4

### PR DESCRIPTION
**Pull request by vimalmanohar**
Fix for a bug that occurs in certain cases.
Also fixes #2269

### CODE REVIEW

**Comment by vimalmanohar** [comment]
This is ready to be merged.

File: egs/babel/s5d/local/run_asr_segmentation.sh [hunk 1]
```diff
@@ -63,13 +63,13 @@ garbage_phones="<oov> <vns>"
 silence_phones="<sss> SIL"
 
 for p in $garbage_phones; do 
-  for affix in "" "_B" "_E" "_I" "_S"; do
+  for a in "" "_B" "_E" "_I" "_S"; do
     echo "$p$affix"
```
Comment (danpovey) [inline egs/babel/s5d/local/run_asr_segmentation.sh]: this doesn't look right to me.

File: egs/aspire/s5/local/run_asr_segmentation.sh [hunk 2]
```diff
@@ -171,21 +159,22 @@ if [ $stage -le 5 ]; then
...
+  steps/segmentation/combine_targets_dirs.sh \
+    $rvb_data_dir ${rvb_targets_dir} \
+    ${rvb_targets_dirs[@]} || exit 1;
+
+  rm -r ${rvb_targets_dirs[@]}
 fi
 
-sad_nnet_dir=exp/segmentation${affix}/tdnn_stats_asr_sad_1a
+sad_nnet_dir=$dir/tdnn_stats_asr_sad_1a
+sad_opts="--extra-left-context 79 --extra-right-context 21 --frames-per-chunk 150 --extra-left-context-initial 0 --extra-right-context-final 0 --acwt 0.3"
```
Comment (danpovey) [inline egs/aspire/s5/local/run_asr_segmentation.sh]: you could remove the comments if they are not relevant any more.
and you should probably explain where you got the "opts" from, e.g. do they need to match something else in another script?

**Comment by vimalmanohar** [comment]
Fixed.

**Comment by danpovey** [comment]
@olix20, can you please check if this resolves your problems?

**Comment by danpovey** [comment]
I'm merging but @olix20, please let us  know if you find further problems.

**Pull request merged**